backend/users/auth.py

```diff
@@ -24,6 +21,8 @@ def downcast_user_type(self, user):
         Retrieve the user from the first table that has a row with same email.
         Subclasses should be checked before parents.
         """
+        if user is None:
+            return None
```

Comment:
in which case 'user' will be None?

Reply:
AnonymousUser. When no one is logged in.